lib/masamune/commands/shell.rb

```diff
@@ -47,8 +47,14 @@ def replace(opts = {})
       around_execute do
         pid = Process.fork
         if pid
-          detach if opts.fetch(:detach, true)
-          Process.waitpid(pid)
+          Signal.trap('INT') {} # Ensure SIGINT is handled by child process exec
```

Comment:
this code is [following double fork pattern](http://stackoverflow.com/questions/1740308/create-a-daemon-with-double-fork-in-ruby)  to prevent interference from parent process. Added a SIGINT ignore so that the child receives SIGINT- since the child might be linked against readline with Ctl-C being a "clear line"